File: Warehousing.Data/Models/PrintSettings.cs
namespace Warehousing.Data.Models
{
    /// <summary>
    /// Print settings structure stored as JSON in User.PrintHeader and User.PrintFooter
    /// This allows flexible customization of print templates
    /// </summary>
    public class PrintHeaderSettings
    {
        /// <summary>
        /// Custom text/html to display at the top of print documents (before company header)
        /// </summary>
        public string? CustomText { get; set; }

        /// <summary>
        /// Show/hide specific elements
        /// </summary>
        public PrintHeaderVisibility Visibility { get; set; } = new PrintHeaderVisibility();
    }

    public class PrintHeaderVisibility
    {
        public bool ShowCompanyName { get; set; } = true;
        public bool ShowCompanyLogo { get; set; } = true;
        public bool ShowCompanyAddress { get; set; } = true;
        public bool ShowCompanyPhone { get; set; } = true;
        public bool ShowCompanyFax { get; set; } = true;
        public bool ShowCompanyEmail { get; set; } = true;
        public bool ShowRegistrationNumber { get; set; } = true;
        public bool ShowCapital { get; set; } = true;
        public bool ShowTaxNumber { get; set; } = true;
        public bool ShowSlogan { get; set; } = true;
        public bool ShowDocumentTitle { get; set; } = true;
    }

    public class PrintFooterSettings
    {
        /// <summary>
        /// Custom text/html to display at the bottom of print documents (after terms and signatures)
        /// </summary>
        public string? CustomText { get; set; }

        /// <summary>
        /// Show/hide specific footer elements
        /// </summary>
        public PrintFooterVisibility Visibility { get; set; } = new PrintFooterVisibility();

        /// <summary>
        /// Custom terms and conditions (overrides company terms if provided)
        /// </summary>
        public string? CustomTerms { get; set; }

        /// <summary>
        /// Custom notes/instructions
        /// </summary>
        public string? CustomNotes { get; set; }
    }

    public class PrintFooterVisibility
    {
        public bool ShowTerms { get; set; } = true;
        public bool ShowNotes { get; set; } = true;
        public bool ShowCustomerSignature { get; set; } = true;
        public bool ShowAuthorizedSignature { get; set; } = true;
        public bool ShowCompanyFooterNote { get; set; } = true;
        public bool ShowDocumentGenerationDate { get; set; } = true;
    }
}

File: Warehousing.UI/src/app/admin/models/PrintSettings.ts
export interface PrintHeaderSettings {
  customText?: string | null;
  visibility: PrintHeaderVisibility;
}

export interface PrintHeaderVisibility {
  showCompanyName: boolean;
  showCompanyLogo: boolean;
  showCompanyAddress: boolean;
  showCompanyPhone: boolean;
  showCompanyFax: boolean;
  showCompanyEmail: boolean;
  showRegistrationNumber: boolean;
  showCapital: boolean;
  showTaxNumber: boolean;
  showSlogan: boolean;
  showDocumentTitle: boolean;
}

export interface PrintFooterSettings {
  customText?: string | null;
  customTerms?: string | null;
  customNotes?: string | null;
  visibility: PrintFooterVisibility;
}

export interface PrintFooterVisibility {
  showTerms: boolean;
  showNotes: boolean;
  showCustomerSignature: boolean;
  showAuthorizedSignature: boolean;
  showCompanyFooterNote: boolean;
  showDocumentGenerationDate: boolean;
}

// Default values
export const defaultPrintHeaderVisibility: PrintHeaderVisibility = {
  showCompanyName: true,
  showCompanyLogo: true,
  showCompanyAddress: true,
  showCompanyPhone: true,
  showCompanyFax: true,
  showCompanyEmail: true,
  showRegistrationNumber: true,
  showCapital: true,
  showTaxNumber: true,
  showSlogan: true,
  showDocumentTitle: true
};

export const defaultPrintFooterVisibility: PrintFooterVisibility = {
  showTerms: true,
  showNotes: true,
  showCustomerSignature: true,
  showAuthorizedSignature: true,
  showCompanyFooterNote: true,
  showDocumentGenerationDate: true
};






